server/pbenchinacan/run-pbench-in-a-can

```diff
@@ -8,17 +28,99 @@ export PB_ELASTICSEARCH_IMAGE_TAG=${PB_ELASTICSEARCH_IMAGE_TAG:-"7.13.3"}
 export PB_SERVER_IMAGE_TAG=${PB_SERVER_IMAGE_TAG:-"$(cat jenkins/branch.name)"}
 export PB_POD_NAME=${PB_POD_NAME:-"pbench-in-a-can_${PB_SERVER_IMAGE_TAG}"}
 
-export PB_COMMON_IMAGE_PULL_POLICY=${PB_COMMON_IMAGE_PULL_POLICY:-"Always"}
+export PB_COMMON_IMAGE_PULL_POLICY=${PB_COMMON_IMAGE_PULL_POLICY:-"always"}
 export PB_ELASTICSEARCH_IMAGE_PULL_POLICY="${PB_ELASTICSEARCH_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
 export PB_POSTGRESQL_IMAGE_PULL_POLICY="${PB_POSTGRESQL_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
 export PB_KEYCLOAK_IMAGE_PULL_POLICY="${PB_KEYCLOAK_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
 export PB_SERVER_IMAGE_PULL_POLICY="${PB_SERVER_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
 
 export PB_DASHBOARD_DIR="${PB_DASHBOARD_DIR:-${PWD}/dashboard/build/}"
 
-envsubst < $(dirname ${0})/pod.yml | podman play kube -
+podman pod create --name ${PB_POD_NAME} --hostname pbenchinacan -p 9200:9200 -p 5432:5432 -p 8090:8090 -p 8080:8080
+
+podman run \
+  --rm \
+  --name "${PB_POD_NAME}-elasticsearch" \
+  --pod ${PB_POD_NAME} \
+  --workdir /usr/share/elasticsearch \
+  --pull ${PB_ELASTICSEARCH_IMAGE_PULL_POLICY} \
+  --umask 0000 \
+  --restart no \
+  --pids-limit 0 \
+  -e PATH=/usr/share/elasticsearch/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
+  -e HOME=/root \
+  -e TERM=xterm \
+  -e container=oci \
+  -e discovery.type=single-node \
+  -e ES_JAVA_OPTS="-Xms1g -Xmx1g"\
+  -e xpack.security.enabled=false \
+  -e ES_PORT=9200 \
+  -d \
+  ${PB_ELASTICSEARCH_IMAGE_NAME}:${PB_ELASTICSEARCH_IMAGE_TAG} \
+  /bin/tini -- /usr/local/bin/docker-entrypoint.sh eswrapper
+
+
+podman run \
+  --rm \
+  --pod ${PB_POD_NAME} \
+	--name "${PB_POD_NAME}-postgres" \
+	--workdir /opt/app-root/src \
+	--pull ${PB_POSTGRESQL_IMAGE_PULL_POLICY} \
+	--umask 0000 \
+	--user 26:26 \
+	--restart no \
+  --pids-limit 0 \
+  --entrypoint container-entrypoint \
+	-e POSTGRESQL_USER=pbenchcontainer \
+	-e POSTGRESQL_PASSWORD=pbench \
+	-e POSTGRESQL_DATABASE=pbenchcontainer \
+	-e HOME=/var/lib/pgsql \
+	-e CONTAINER_SCRIPTS_PATH=/usr/share/container-scripts/postgresql \
+	-e APP_ROOT=/opt/app-root \
+	-e POSTGRESQL_VERSION="13" \
+	-e PGUSER=postgres \
+	-e STI_SCRIPTS_URL=image:///usr/libexec/s2i \
+	-d \
+	${PB_CONTAINER_REG}/${PB_ORG_NAME}/${PB_POSTGRESQL_IMAGE_NAME}:${PB_POSTGRESQL_IMAGE_TAG} \
+	run-postgresql -p 5432
+
+podman run \
+  --rm \
+  --pod ${PB_POD_NAME} \
+  --name "${PB_POD_NAME}-keycloak" \
+  --pull ${PB_KEYCLOAK_IMAGE_PULL_POLICY} \
+  --umask 0000 \
+  --restart no \
+  --pids-limit 0 \
+  -e KEYCLOAK_ADMIN=admin \
+  -e KEYCLOAK_ADMIN_PASSWORD=admin \
+  -d \
+  quay.io/keycloak/keycloak:latest \
+  start-dev --health-enabled=true --http-port=8090
+
+sh server/pbenchinacan/load_keycloak.sh
+
+# Create a dashboard src directory to mount on pbench-server container's
+# dashboard directory
+mkdir -m 0755 -p ${PB_DASHBOARD_DIR}
```

Comment:
This strikes me as odd:  why would we want to create a local and **empty** directory?

This directory should already exist (and, hopefully, be populated with a build of the Dashboard!) -- if it doesn't, it's likely to cause problems for the running Pbench Server....

If you really want to do something here, then put in a test that it exists (but do the test before starting _any_ containers...).

Reply:
Where do we do that check in our current code right now? We have a pod.yml file where we specify the `dashboard_deployment` volume and we specify the type as `DirectoryOrCreate` which will create that directory if it's not present, I think, right? This file is doing the exact same thing the pod.yml would do as I am not trying to add any more complexities at this point. 

> This directory should already exist (and, hopefully, be populated with a build of the Dashboard!) -- if it doesn't, it's likely to cause problems for the running Pbench Server....

I agree, but in our code today, we are not building pbench-dashboard in our CI server image, are we?

Reply:
> we are not building pbench-dashboard in our CI server image

You are correct that the CI does not copy the Dashboard into the container image:  as you point out, the `pod.yml` file causes it to be mapped in from the local file system when the container is invoked.

> we specify the type as `DirectoryOrCreate` which will create that directory if it's not present

Correct; I don't remember exactly why I selected that option.

On the other hand, applying `mkdir -p` to an existing directory probably does no harm (although, I don't think the `-m` does anything unless `mkdir` actually _creates_ a directory), so the code is presumably OK; however, the comment should be updated to point out that the **expectation** is that the directory already exists and that the `mkdir` invocation is just there to cover the case when the user is deliberately running the Pbench Server without a Dashboard deployment.